plugin/src/main/java/me/champeau/mrjar/MultiReleaseExtension.java

```diff
@@ -28,24 +28,26 @@
 import org.gradle.api.plugins.JavaApplication;
 import org.gradle.api.plugins.JavaPluginExtension;
 import org.gradle.api.plugins.PluginManager;
+import org.gradle.api.plugins.jvm.JvmTestSuite;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.JavaExec;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.SourceSetContainer;
 import org.gradle.api.tasks.TaskContainer;
-import org.gradle.api.tasks.TaskProvider;
 import org.gradle.api.tasks.bundling.Jar;
 import org.gradle.api.tasks.compile.JavaCompile;
-import org.gradle.api.tasks.testing.Test;
 import org.gradle.jvm.toolchain.JavaCompiler;
 import org.gradle.jvm.toolchain.JavaLanguageVersion;
 import org.gradle.jvm.toolchain.JavaLauncher;
 import org.gradle.jvm.toolchain.JavaToolchainService;
 import org.gradle.language.base.plugins.LifecycleBasePlugin;
+import org.gradle.testing.base.TestingExtension;
+import org.gradle.util.GradleVersion;
 
+import java.util.Arrays;
 import javax.inject.Inject;
 
-public abstract class MultiReleaseExtension {
```

Comment:
I'm not sure why this class is abstract, but it'd only make sense for the constructor to have `protected` access, right? However, the constructor must be `public` for the extension to be created, and if no other class is extending this one, it doesn't make sense for it to be abstract.

I can change it back to be abstract. I just wanted to understand the why behind it 🙂

Reply:
It is abstract because it is not supposed to be instantiated directly, but only via Gradle's instantiator mechanism.